lib/chef/azure/commands/enable.rb

```diff
@@ -65,19 +63,26 @@ def enable_chef
     # - Install the Chef service
     # - Start the Chef service
     begin
-      configure_chef_only_once
+      # "node-registered" file also indicates that enabled was called once and
+      # configs are already generated.
+      if File.exist?("#{bootstrap_directory}/node-registered")
+        puts "#{Time.now} Node is already registered..."
+      else
+        configure_chef_only_once
+      end
+
+      @daemon = value_from_json_file(handler_settings_file, 'runtimeSettings', '0', 'handlerSettings', 'publicSettings', 'daemon')
+      @daemon = (@daemon.nil? || @daemon.empty? || @daemon == "service") ? "service" : "task"

```

Comment:
`"daemon": "none"` provided as a setting should cause us to just install Chef Client and do the first run, but not set up as a service or task - this line looks like it might cause a task to be created?

Reply:
True. Will update this. Thanks